wordpress-mobile/WordPress-Android
# Add informative Gravatar UI to the profile editing

Fixes #20173

-----

We want to inform users that when editing a profile, they are actually editing their Gravatar profile. In this PR, we are adding an informative view for that.

We've created a similar [PR](https://github.com/wordpress-mobile/WordPress-iOS/pull/22563) on the iOS repo to keep both apps on a similar page.

The difference with the iOS app is that we'll show the Gravatar information banner in `MeFragment` instead of `MyProfileFragment`. That's because, in the Android app, users modify their avatar on the first screen, so we need to show the notice banner on that screen. 

Sync info banner will be shown in both fragments after any modification (avatar or field) is applied.

Jetpack:

| Dark Mode | Light Mode | Dark Mode Sync Message | Light Mode Sync Message | Dark - Profile Sync | Light - Profile Sync |
|----------|-----------|----------|-----------|----------|-----------|
|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/0a3d0e87-9bbd-4e99-a60d-75c838477a9e">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/f0acc9b6-781f-49b1-abd5-907f2fdc183a">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/cc7fa030-ed7c-4204-81cc-292b5366392b">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/49ead560-aa3c-44ce-a74f-e415fdfbb3c8">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/f2e41964-bf7b-42e6-ba3a-9327e57e04e2">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/2165aa89-eba8-4cf0-8ece-2010282dfc83">|

WordPress:

| Dark Mode | Light Mode | Dark Mode Sync Message | Light Mode Sync Message | Dark - Profile Sync | Light - Profile Sync |
|----------|-----------|----------|-----------|----------|-----------|
|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/2192601d-4ab7-4f53-b91f-3eaf1f78f792">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/c01a5685-6abc-4ec7-9518-dec0357eed82">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/625cc5b9-7e6f-4f4f-85c1-b951dcdbc694">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/2796ff7c-7d79-4bc1-863b-648c10886d4d">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/c96911a1-5cdc-4a31-8211-acba5b1157e3">|<img width="250" alt="image" src="https://github.com/wordpress-mobile/WordPress-Android/assets/6974554/9854e34d-a989-4cd0-8d14-ebad14e27ff4">|


## To Test:

Try on both JP and WP:

- Go to the `Me tab`
- Observe: There's a new informative UI as demonstrated above ☝️ (_If possible, verify in dark and light mode_)
- Click the link `Learn more on Gravatar.com` and verify you are redirected to `www.gravatar.com`
- Go back to the App
- Update your avatar
- Observe you see a notice message saying "Updates might take some time to sync with your Gravatar profile."
- Click the `Done` button and verify the notice message disappears
- Open `My Profile`
- Try updating those 4 fields
- Observe you see a notice message saying "Updates might take some time to sync with your Gravatar profile."
- Click the `Done` button and verify the notice message disappears

-----

## Regression Notes

1. Potential unintended areas of impact

    - We shouldn't impact other areas. We are adding views in `MyProfileFragment` and `MeFragment`, so all possible impacts should be inside that fragment.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - N/A

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR Submission Checklist:


-----

## Testing Checklist:
